src/chapters/http/web-addresses.rst

```diff
@@ -0,0 +1,123 @@
+   - ``https://en.wikipedia.org/wiki/Client–server_model``
+   - ``https://repl.it``
+
+A URL contains a lot of information about the request, including *what* is
+being requested and *where* it should be sent. URLs contain several parts, and
+the general structure looks like:
+
+::
+
+   scheme://host:port/path?query_string
+
+The five parts of this URL are:
+
+#. Scheme
+#. Host
+#. Port (optional)
+#. Path (optional)
+#. Query String (optional)
+
+Only the ``scheme`` and ``host`` are required, and the parts must be in the
+*exact* order shown. Let's look at each of these in detail.
+
+Scheme
+------
+
+.. index::
+   single: url; scheme
+
+The first portion of every URL specifies the **scheme**. Common schemes include
+``http``, ``https``, and ``file``. The scheme usually identifies the *protocol*
+for the request. For us, this will always be ``http`` or ``https``. If left
+off, a web browser will insert the scheme http/s for you. 
+
+The scheme is *always* followed by ``://``.
+
+Host
+----
+
+.. index::
+   single: url; domain
+
+The **host** portion of a URL specifies where the request should be sent. The
+host can be either an IP address, like ``104.25.128.113``, or a domain name,
+like ``www.launchcode.org``.
+
+Port
+----
+
+.. index::
+   single: url; port
+
+Following the host is an optional **port** number. The port determines which
+application on the server needs to handle the request.
+
+When the port number is missing, a default value is used. For ``http://``, the
+default port is ``80``. When using ``https://``, the default port is ``443``.
+
+Path
+----
+
+.. index::
+   single: url; path
+
+The **path** of a URL consists of a series of names separated by ``/``. It
+looks similar to the result returned when we enter the ``pwd`` command in the
+terminal.
+
+The path tells the server *what* is being requested. It includes a series of
+names that end with a specific directory or file. For example,
+``/blog/entries/2018/`` requests the contents of the ``2018`` directory, while
+``/blog/index.html`` requests only the ``index.html`` file from the ``blog``
+folder.
+
+If the path is missing in the URL, then the **root path** ``/`` is used. The
+root path usually sends us to the home page for a given site.
+
+Query String
+------------
+
+.. index::
+   single: url; query string
+
+After the path comes the **query string**, which begins with ``?``. The *query*
+itself contains a set of key-value pairs. Each pair is joined by ``=`` and is
+separated from the other pairs by ``&``.
+
+While the path specifies *what* the request is asking for, the query string
+provides additional information needed to fulfill the request. For example, a
+query string from a search using DuckDuckGo might look like this:
+
+::
+
+   ?q=recent+nasa+images&ia=images
+
+This query has *two* key-value pairs:
+
+- ``q`` : ``recent+nasa+images``
+- ``ia`` : ``images``
+
+A search for "NASA" returns its `home page <https://www.nasa.gov/>`__ and
```

Comment:
Explicitly including a link to the search results page would allow students to see the query string in the address bar, as well as the results filtered down by the query params.

Reply:
I updated the link.